src/dotnet/Core.Server/ServiceCollectionExt.cs

```diff
@@ -119,4 +130,43 @@ public static IServiceCollection AddNats(this IServiceCollection services, HostI
         services.AddSingleton(optionsBuilder ?? (static _ => new NatsQueues.Options()));
         return services;
     }
+
+    // EntityResolver
+
+    public static IServiceCollection OverrideEntityResolver(this IServiceCollection services)
+    {
+        var serviceResolverDescriptors = services
+            .Where(sd => sd.ServiceType.IsGenericType && sd.ServiceType.GetGenericTypeDefinition() == typeof(IDbEntityResolver<,>))
+            .ToList();
+        foreach (var serviceDescriptor in serviceResolverDescriptors) {
+            var existingFactory = serviceDescriptor.ImplementationFactory;
+            if (existingFactory == null)
+                continue;
+
+            var implementationType = ServiceDescriptorGetImplementationType(serviceDescriptor);
+            if (implementationType == null)
+                continue;
+
+            if (!implementationType.IsGenericType)
+                continue;
+
+            if (implementationType.GetGenericTypeDefinition() != typeof(IDbEntityResolver<,>))
+                continue;
+
+            var factory = _factoryMap.GetOrAdd(implementationType,
```

Comment:
Do we get any improvment from using `_factoryMap`? I think implementation type will be unique in every usage.